src/budget.rs

```diff
@@ -189,4 +215,14 @@ mod tests {
         budget.apply_witness(&Witness::Signature, &from);
         assert_eq!(budget, Budget::new_payment(42, from));
     }
+    #[test]
+    fn test_dual_authorized_payment() {
+        let from0 = Keypair::new().pubkey();
+        let from1 = Keypair::new().pubkey();
+        let to = Pubkey::default();
+
+        let mut budget = Budget::new_dual_authorized_payment(from0, from1, 42, to);
+        budget.apply_witness(&Witness::Signature, &from0);
+        assert_eq!(budget, Budget::new_authorized_payment(from1, 42, to));
```

Comment:
This is perfect. Glad to see it goes no further than what's needed to test the new code.